### ISSUE TITLE
lots of warnings about deprecated decalarations when building

### ISSUE THREAD

**Issue by filochard** [openmonk/guayadeque#3]
I built a rpm of guayadeque with a snapshot from november 21st
It works perfectly :-)

I only have warnings about deprecated declarations when building  but they don't prevent to build and Guayadeque works normally despite of them

I attach  here  extracts of the buildlog if it may be useful for you : 
[build-log-extracts.txt](https://github.com/openmonk/guayadeque/files/13464485/build-log-extracts.txt)

It may be easier with an index for this extract : 

line 1                   'GstPad* gst_element_get_request_pad(GstElement*, const gchar*)' is deprecated:
 src/audio/FaderPlaybin 
deprecated declaration  to gstreamer 
 (I attach a patch for this here : it's OK : it suppresses the warning when building and Guayadeque works as well)
[FaderPlaybin.txt](https://github.com/openmonk/guayadeque/files/13464504/FaderPlaybin.txt)

Line 16               wxFileName::Normalize is deprecated
/src/db/DbLibrary.cpp:3333

Line 32 
/src/misc/MD5.cpp:183
ISO C++17 does not allow 'register' storage class specifier [-Wregister]

Line 48 to 344
 /src/taginfo/TagInfo.cpp
several  warnings about XiphComment::removeField(const TagLib::String&, const TagLib::String&) being deprecated
or  TAGLIB_DEPRECATED ItemMap &itemListMap();

Line 348            Reader' is deprecated: Use CharReader and CharReaderBuilder instead.
src/ui/cover/Apple.cpp:52

_____________________________________________________________________________________________________________
following warnings are about deprecated wxPath or wxFileName or wxFont

Line 366 to 384           'wxPATH_NORM_ALL' is deprecated
/src/ui/magnatune/Magnatune.cpp:443
/src/ui/magnatune/Magnatune.cpp:1085

Line 387 to 405           'wxPATH_NORM_ALL' is deprecated
/src/ui/lyrics/LyricsPanel.cpp:1802
/src/ui/lyrics/LyricsPanel.cpp:1903

Line 409 to 424           'wxPATH_NORM_ALL' is deprecated
/src/ui/mediaviewer/playlists/PlayListFile.cpp:234
/src/ui/mediaviewer/playlists/PlayListFile.cpp:310

Line 429 to 447         'wxFileName::Normalize()' is deprecated:
/src/ui/mediaviewer/portablemedia/PortableMedia.cpp:556
/src/ui/mediaviewer/portablemedia/PortableMedia.cpp:604

Line  449 to 466            'wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)' is deprecated: 
/src/ui/mediaviewer/portablemedia/PortableMedia.cpp:699
/src/ui/mediaviewer/portablemedia/PortableMedia.cpp:708

Line 471 to 479            'wxPATH_NORM_ALL' is deprecated:
/src/ui/podcasts/ChannelEditor.cpp:76
/src/ui/podcasts/ChannelEditor.cpp:24

Line 481 to 505             'wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)' is deprecated: 
/src/ui/podcasts/ChannelEditor.cpp:104
/src/ui/podcasts/ChannelEditor.cpp:116
/src/ui/podcasts/ChannelEditor.cpp:126

Line 510 to 532             'wxPATH_NORM_ALL' is deprecated:
/src/ui/podcasts/Podcasts.cpp:216
/src/ui/podcasts/Podcasts.cpp:492
/src/ui/podcasts/Podcasts.cpp:698

Line 536 to 597               'wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)' is deprecated:  
/src/ui/podcasts/PodcastsPanel.cpp:946
/src/ui/podcasts/PodcastsPanel.cpp:952
/src/ui/podcasts/PodcastsPanel.cpp:962
/src/ui/podcasts/PodcastsPanel.cpp:972
/src/ui/podcasts/PodcastsPanel.cpp:983
/src/ui/podcasts/PodcastsPanel.cpp:998
/src/ui/podcasts/PodcastsPanel.cpp:1008
/src/ui/podcasts/PodcastsPanel.cpp:1018
/src/ui/podcasts/PodcastsPanel.cpp:1028

Line 599 to the end         'wxPATH_NORM_ALL' is deprecated:
/src/ui/podcasts/PodcastsPanel.cpp:1215
/src/ui/podcasts/PodcastsPanel.cpp:1281
/src/ui/podcasts/PodcastsPanel.cpp:1484

**Comment by thothix** [comment]
Hi @filochard 

I removed the warnings and did some other changes. That's in my repo if you are interested, @tiagobar

**Comment by filochard** [comment]
I'm gonna test what you brought to the source

You might propose these commits to openmonk unless you want to continue to work on your fork
or even contact anonbeat as he said : 
"If anyone wishes to continue the development and support of Guayadeque, please feel free to contact me"
( the openmonk's fork is not indicated on anonbeat site and not so easy to know)

**Comment by filochard** [comment]
Hi tiagobar
I have built and tested your fork (last commit from 2024-09-11)
The build went  well : no more warning when building (even with gcc14)
I installed it and have run it using my previous /home/myusername/.guayadeque (that I have copied as .guayadequebak in case of a problem)
It's a huge database of 19 000 items
It works perfectly 

Post Scriptum
you should add an issue tab to your git repo or a way to communicate outside of the openmonk repo

**Comment by thothix** [comment]
I started this fork based on @NullPointerAchitect version and he based on @openmonk fork. As @NullPointerArchitect archived his code I wasn't able to send him my updates. @openmonk still have opened pull requests without feedback since march so I decided to start a fork to do some fixes to be able to run in my machine.

I don't know yet if I'll be able to maintain Guayadeque, mainly the more advanced pieces of code.
If I think it'll be possible I can reach out @anonbeat to ask him about continue the development officially. Let's see.
By now it's only a person and unofficial Guayadeque fork like the others.

Greets

**Comment by filochard** [comment]
Hi
I only knew openmonk's fork
didn't know where to find others
I discovered yours with your post on openmonk's github (thanks for this idea)

**Comment by thothix** [comment]
> Hi I only knew openmonk's fork didn't know where to find others I discovered yours with your post on openmonk's github (thanks for this idea)

You can find @NullPointerArchitect in the 4 opened pull requests here. I opened an issue in my repo.

**Comment by thothix** [comment]
@filochard, @raschaa
I changed my github profile:

https://github.com/thothix

**Comment by raschaa** [comment]
Yeah, I noticed, but no prob, github steered me in the right direction anyway...
